lib/elixir/lib/kernel.ex

```diff
@@ -263,7 +263,7 @@ defmodule Kernel do
       1
 
   """
-  @spec hd(maybe_improper_list) :: term
+  @spec hd(nonempty_maybe_improper_list(elem, term)) :: elem when elem: var
```

Comment:
I would prefer to stick to one, since we have `var`, `term` and `any`. is there a reason when to use `var` over the other ones?